### PR TITLE
Add pkcs#11 3.0 mock modules for testing

### PR DESCRIPTION
**Pull request by ZoltanFridrich**
Adds new mock modules for testing purposes of pkcs#11 3.0

### CODE REVIEW

File: p11-kit/mock-module-v3-ep4.c [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+static CK_RV
+override_initialize (CK_VOID_PTR init_args)
+{
+	if (init_pid != getpid ())
+		return CKR_GENERAL_ERROR;
+	return mock_C_Initialize (init_args);
+}
+
+/* Present for backward compatibibility */
```
Comment (ueno) [inline p11-kit/mock-module-v3-ep4.c:57]: typo: "compatibility"

File: p11-kit/mock-module-v3-ep.c [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+	for (i = 0; i < MOCK_INTERFACES; i++) {
+		/* Version is the first member of CK_FUNCTION_LIST */
+		CK_VERSION_PTR interface_version = (CK_VERSION_PTR)mock_interfaces[i].pFunctionList;
+
+		/* The interface name is not null here */
+		if (strcmp ((char *)pInterfaceName, mock_interfaces[i].pInterfaceName) != 0) {
+			continue;
+		}
+		/* If version is not null, it must match */
+		if (pVersion != NULL_PTR && (pVersion->major != interface_version->major ||
```
Comment (ueno) [inline p11-kit/mock-module-v3-ep.c:126]: Wouln't it be simpler to write: `*pVersion != *interface_version`?

**Review by ueno — APPROVED** [review]

File: p11-kit/mock-module-v3-ep.c (hunk 2 above)
Comment (ZoltanFridrich) [reply p11-kit/mock-module-v3-ep.c]: actually, compiler has some problems with it for some reason

Comment (ZoltanFridrich) [reply p11-kit/mock-module-v3-ep.c]: I tried to retype it and the compiler hates it no matter what. I am keeping it as it was originally.

**Pull request merged**